verifier/testreport.py

```diff
@@ -198,29 +139,57 @@ def summary_status(self):
     return self.tests_fail == 0 and not self.missing_verify_data
 
   def compute_category_summary(self, items, group_tag, detail_tag):
-    # TODO: remove this early exit
-    return
-    
     # For the items, count messages and arguments for each
     groups = {}
     for item in items:
-      print('@@@@@@@@@@@ item %s' % item)
-      group = item.get(group_tag)
-      detail = item.get(detail_tag)
+      label = item['label']
+      details = item.get('error_detail')
+      if not details:
+        # Try getting the group_tag
+        details = item.get(group_tag)
+      if isinstance(details, str):
+        detail = details
+        group = group_tag
+      else:
+        detail = details.get(group_tag)
+        group = group_tag
+
       if group:
-        if not groups.get(group):
-          groups[group].append(item)
+        if groups.get(group):
+          groups[group][str(detail)].append(label)
         else:
-          groups[group]= [item]
-    print('GROUPS FOR %s %s = \n%s' % (group_tag, detail_tag, groups.keys()))
+          groups[group]= {str(detail): [label]}
+    # print('GROUPS FOR %s %s = \n%s' % (group_tag, detail_tag, groups.keys()))
+    return groups
+
+  def compute_unsupported_category_summary(self, items, group_tag, detail_tag):
+    # For the items, count messages and arguments for each
+    groups = {}
+    for item in items:
+      error_detail = item.get('error_detail')
```

Comment:
Rename local variable identifiers.
Also consider renaming key in test executor result structs from `error_detail` to something more generic if errors and unsupported exceptions can both occur.